src/main/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTask.java

```diff
@@ -17,55 +17,114 @@
+import com.google.cloud.tools.managedcloudsdk.command.CommandExecutionException;
+import com.google.cloud.tools.managedcloudsdk.command.CommandExitException;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponent;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponentInstaller;
+import com.google.cloud.tools.managedcloudsdk.install.SdkInstaller;
+import com.google.cloud.tools.managedcloudsdk.install.SdkInstallerException;
+import com.google.cloud.tools.managedcloudsdk.update.SdkUpdater;
+import java.io.IOException;
 import org.gradle.api.DefaultTask;
-import org.gradle.api.GradleException;
 import org.gradle.api.tasks.TaskAction;
 
 public class DownloadCloudSdkTask extends DefaultTask {
 
-  private CloudSdkDownloader downloader;
   private CloudSdkBuilderFactory cloudSdkBuilderFactory;
-  private ToolsExtension toolsExtension;
-
-  public void setToolsExtension(ToolsExtension toolsExtension) {
-    this.toolsExtension = toolsExtension;
-  }
+  private ManagedCloudSdkFactory managedCloudSdkFactory;
+  private ManagedCloudSdk managedCloudSdk;
 
   public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFactory) {
     this.cloudSdkBuilderFactory = cloudSdkBuilderFactory;
   }
 
-  public void setSdkDownloader(CloudSdkDownloader downloader) {
-    this.downloader = downloader;
+  public void setManagedCloudSdkFactory(ManagedCloudSdkFactory managedCloudSdkFactory) {
+    this.managedCloudSdkFactory = managedCloudSdkFactory;
   }
 
   /** Task entrypoint : Download/update/verify Cloud SDK installation. */
   @TaskAction
-  public void downloadCloudSdkAction() {
-    String sdkVersion = toolsExtension.getCloudSdkVersion();
-    File sdkHome = toolsExtension.getCloudSdkHome();
-
-    if (sdkHome == null) {
-      if (Strings.isNullOrEmpty(sdkVersion)) {
-        // Wants to download, but version isn't specified; assume latest version
-        sdkVersion = "LATEST";
-      }
-      sdkHome = downloader.downloadSdk(sdkVersion);
-    } else {
-      if (!Strings.isNullOrEmpty(sdkVersion)) {
-        // Sdk home and version specified; validate installation
-        if (!downloader.isSdkValid(sdkVersion, sdkHome)) {
-          throw new GradleException(
-              "Specified Cloud SDK version and actual version of the SDK installed in the "
-                  + "specified directory do not match. You must either specify the correct "
-                  + "cloudSdkHome and cloudSdkVersion, or you can remove the cloudSdkHome field "
-                  + "to download the version you want.");
-        }
-      }
+  public void downloadCloudSdkAction()
+      throws UnsupportedOsException, BadCloudSdkVersionException, ManagedSdkVerificationException,
+          ManagedSdkVersionMismatchException, InterruptedException, CommandExecutionException,
+          SdkInstallerException, CommandExitException, IOException {
+    managedCloudSdk = managedCloudSdkFactory.newManagedSdk();
```

Comment:
This looks like it could be a local variable?

Reply:
Fixed